src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -277,68 +251,53 @@ def random_storage_account_name(registry_name):
 
 
 def managed_registry_validation(registry_name, resource_group_name=None, message=None):
-    """Validates if a registry is in Basic SKU.
+    """Raise CLIError if the registry in not in Managed SKU.
     :param str registry_name: The name of container registry
     :param str resource_group_name: The name of resource group
     """
-    arm_resource = _arm_get_resource_by_name(registry_name, ACR_RESOURCE_TYPE)
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
 
-    if arm_resource.sku.tier == SkuTier.basic.value:
-        raise CLIError(message or "This operation is not supported for registries in Basic SKU.")
+    if not registry.sku or registry.sku.name not in MANAGED_REGISTRY_SKU:
+        raise CLIError(message or "This operation is only supported for managed registries.")
 
-    if not resource_group_name:
-        resource_group_name = get_resource_group_name_by_resource_id(arm_resource.id)
+    return registry, resource_group_name
+
+
+def premium_registry_validation(registry_name, resource_group_name=None, message=None):
+    """Raise CLIError if the registry in not in Premium SKU.
+    :param str registry_name: The name of container registry
+    :param str resource_group_name: The name of resource group
+    """
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
+
+    if not registry.sku or registry.sku.name != SkuName.premium.value:
+        raise CLIError(message or "This operation is only supported for managed registries in Premium SKU.")
 
-    return arm_resource, resource_group_name
+    return registry, resource_group_name
 
 
-def validate_sku_update(sku_parameter):
+def validate_sku_update(current_sku, sku_parameter):
     """Validates a registry SKU update parameter.
     :param object sku_parameter: The registry SKU update parameter
     """
     if sku_parameter is None:
         return
 
     if isinstance(sku_parameter, dict):
-        if 'name' not in sku_parameter or sku_parameter['name'] not in MANAGED_REGISTRY_SKU:
+        if 'name' not in sku_parameter:
             _invalid_sku_update()
+        if current_sku in MANAGED_REGISTRY_SKU and sku_parameter['name'] in CLASSIC_REGISTRY_SKU:
+            _invalid_sku_downgrade()
     elif isinstance(sku_parameter, Sku):
-        if sku_parameter.name not in MANAGED_REGISTRY_SKU:
-            _invalid_sku_update()
+        if current_sku in MANAGED_REGISTRY_SKU and sku_parameter.name in CLASSIC_REGISTRY_SKU:
+            _invalid_sku_downgrade()
     else:
         _invalid_sku_update()
 
 
 def _invalid_sku_update():
```

Comment:
It may not be necessary to have a method just throw an exception. You can extract the error message.

Reply:
I do agree that client side validation should be avoided because it could introduce inconsistency with server side and more maintenance effort, but we do also noticed that certain client side validation can improve usability by rejecting the call immediately rather than sending the request and getting back a 400 with a few seconds delay. The same is true for `validate_managed_registry` and `validate_premium_registry`.